File: v1/src/main/java/com/api/v1/creditcar/CreditCard.java
package com.api.v1.creditcar;

import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDate;

@Document(collection = "v1_credit_cards")
public record CreditCard(
    String ownerName,
    String cardNumber,
    String flag,
    String cvc,
    LocalDate dueDate,
    String addedAt
) {}

File: v2/src/main/kotlin/com/api/v2/creditcard/CreditCard.kt
package com.api.v2.creditcard

import org.springframework.data.mongodb.core.mapping.Document
import java.time.LocalDate

@Document(collection = "v1_credit_cards")
data class CreditCard(
    val ownerName: String,
    val cardNumber: String,
    val flag: String,
    val cvc: String,
    val dueDate: LocalDate,
    val addedAt: String
)
